plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/DBNResource.java

```diff
@@ -216,7 +216,11 @@ public boolean allowsChildren() {
     @Override
     public DBNNode[] getChildren(DBRProgressMonitor monitor) throws DBException {
         if (children == null) {
-            this.children = readChildNodes(monitor);
+            synchronized (this) {
```

Comment:
I don't think that we can use sync here. It may lead to deadlocks because readChildNodes may interact with UI.